### ISSUE TITLE
Migrate to flutters code generation

### ISSUE THREAD

**Issue by szotp** [timfreiheit/r_flutter#1]
Hi, a while ago this was implemented in flutter:
https://github.com/flutter/flutter/wiki/Code-generation-in-Flutter

Perhaps this project could be migrated to use it, this way `flutter packages pub run` would not be needed anymore.

I forked this project already and I'm going to see if its doable.

**Comment by timfreiheit** [comment]
Hi,
this would be an interesting oportunity.
I will also try to take a look at it next week.
One feature i can not see in the new solution is passing arguments or options to the code generator.

**Comment by szotp** [comment]
Maybe a config file would do the job? Something like build.yaml but added to the app itself.

**Comment by szotp** [comment]
Alright, I got it to work. There must be an assert.yaml file or something like that, which can contain configuration:
https://github.com/szotp/r_flutter/blob/master/example/lib/assets.yaml

And then this builder will convert it into assets.dart. assets.dart is not in the repo because it's created in the build cache.
https://github.com/szotp/r_flutter/blob/master/lib/builder.dart

I'm not yet sure how to do localization strings.

**Comment by timfreiheit** [comment]
I would love to integrate this feature. I will take a look at this in the evening.

**Comment by timfreiheit** [comment]
looks nice.
one problem I see is that the file is not regenerated when assets are updated.
this might be a limitation on the code generator api but it would require making random changes to the `assets.yaml` to regenerate `assets.dart`

**Comment by szotp** [comment]
It's possible to create a builder that reacts to changes inside lib: 
https://github.com/dart-lang/build/blob/master/docs/writing_an_aggregate_builder.md#writing-the-builder

I haven't tested it yes so I'm not sure how well it would work, it sounds like it would also run every time user saves .dart file, which is not ideal for performance. 

Regarding intl_translation, I don't care much about that code generation. It's super complicated and the end result is source files that still create a Map, it doesn't event generate a delegate. Personally I would get rid of intl completely.

**Comment by timfreiheit** [comment]
that sounds exactly what we need.
We need to calculate the folder patterns based on the asset patterns defined in `pubspec.yaml`.

I would also be fine in getting rid of `intl_translation` in favour of an own code generation. but this can also be deferred

**Comment by szotp** [comment]
Apperently I just had to call `buildStep.findAssets(Glob("**/*"))` for asset reload to work. I will optimize this a bit to only observe directories in pubspec.yaml and submit a PR

**Comment by timfreiheit** [comment]
when we observe the whole assets anyway do you think it would make sense to move the configuration from the `assets.yaml` into the `pubspec.yaml` like

```yaml
r_flutter:
  intl: lib/i18n/en.arb
  ignore:
    - assets/sub/ignore1 #use ignore option to skip 
    - assets/sub/ignore2
    - lib/i18n
```

this would remove the need to maintain different places of configuration

**Comment by szotp** [comment]
Yes, that would be great. I will check if it can be done.

**Comment by szotp** [comment]
I got it to work, thanks. I think it's ready so I created the PR.

**Comment by timfreiheit** [comment]
the package was released as the version `0.2.1` now https://pub.dev/packages/r_flutter
This repository was also renamed to `r_flutter` to make is consistent.
Thank you very much.
